### PR TITLE
Fix #532 and the comment Andre adds 

### LINKED ISSUE #532: Backspace of quote/bracket in REPL window doesn't remove auto completed pair
In the editor, it works correctly, if you type "|" and then hit backspace.  [Pipe used to represent Carat]

In the REPL window, if you type "|" and hit backspace, you're left with "

### PR DESCRIPTION
**Pull request by HuanhuanSunMSFT**
With this fix, the behavior would be:
- Braces always get auto completed unless in string or comment area
- Quotes get auto completed when there is no text succeeding the caret in the same line or when it's typed right after an auto-completion
  @AndreSayreMSFT @EricMSFT @Microsoft/poshtools

### CODE REVIEW

File: PowerShellTools/Intellisense/Utilities.cs [hunk 1]
```diff
@@ -147,25 +147,25 @@ internal static int GetCurrentBufferPosition(ITextView textView, out ITextBuffer
...
         }
 
         internal static bool IsInVariableArea(int position, ITextBuffer buffer)
         {
-            return IsInCertainPSTokenTypesArea(position, buffer, PSTokenType.Variable, PSTokenType.Member);
+            return IsInCertainPSTokenTypesArea(position, buffer, true, PSTokenType.Variable, PSTokenType.Member);
         }
 
-        private static bool IsInCertainPSTokenTypesArea(int position, ITextBuffer buffer, params PSTokenType[] selectedPSTokenTypes)
+        private static bool IsInCertainPSTokenTypesArea(int position, ITextBuffer buffer, bool edgeIncluded, params PSTokenType[] selectedPSTokenTypes )
```
Comment (ASayre) [inline PowerShellTools/Intellisense/Utilities.cs:168]: Want to include a default for edgeIncluded?

Comment (HuanhuanSunMSFT) [inline PowerShellTools/Intellisense/Utilities.cs:168]: Doesn't make much difference as you have to specify the value when calling the method. But I'll default it to true as an indicator that true is more common.

Comment (ASayre) [inline PowerShellTools/Intellisense/Utilities.cs:168]: Ah, can scratch, didn't catch the fact that the last parameter set was a 'params'

**Comment by ASayre** [comment]
Looks good

File: PowerShellTools/Intellisense/Utilities.cs (hunk 1 above)
Comment (HuanhuanSunMSFT) [inline PowerShellTools/Intellisense/Utilities.cs:168]: NP. Then I'll just leave it as it is.:)

**Pull request merged**